src/server/prepare.js

```diff
@@ -20,7 +20,16 @@ function prepare() {
 
         lastPlatform = platform;
 
-        exec('cordova prepare ' + platform, function (err, stdout, stderr) {
+        var projectRoot;
+        try {
+            projectRoot = config.projectRoot;
+        } catch (error) {
+            return Q.reject(error);
```

Comment:
The reason of returning an already rejected promise here is to avoid the execution flow until the exec call.